spinalcordtoolbox/scripts/sct_apply_transfo.py

```diff
@@ -336,25 +336,16 @@ def apply(self):
 
 # MAIN
 # ==========================================================================================
-def main(args=None):
+def main(argv=None):
     """
     Entry point for sct_apply_transfo
-    :param args: list of input arguments. For parameters -w and -winv, args list should include a nested list for every
-    item. Example: args=['-i', 'file.nii', '-w', ['warp1.nii', 'warp2.nii']]
+    :param argv: list of input arguments.
```

Comment:
Why remove the example? is the nested list no longer needed ?

Reply:
Yes, exactly. To make nested lists work, there was some extra parsing logic outside of `argparse`.

https://github.com/neuropoly/spinalcordtoolbox/blob/e07cd93cdd9f856b326f81efdd470d10ea642af2/spinalcordtoolbox/scripts/sct_apply_transfo.py#L347-L355

I removed this extra parsing logic (and the example). I did this for two reasons:

* I felt that all parsing behavior should be handled only inside `argparse`.
* I felt the usage would be clearer if the input formatting was exactly same for main as it is for the CLI parsing. 
    * CLI: `-w warp1.nii warp2.nii` 
    * main: `argv=['-w', 'warp1.nii', 'warp2.nii']`

Reply:
Besides, the nested list usage was only used in 1 place, and that is quite easy to fix. (See https://github.com/neuropoly/spinalcordtoolbox/pull/3091#discussion_r541946975 for more info.)